java-squid/src/main/java/org/sonar/java/se/SymbolicValueRelation.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.se;
+
+import javax.annotation.Nullable;
+
+import java.util.Collection;
+
+public abstract class SymbolicValueRelation {
+
+  public static EqualRelation equal(SymbolicValue v1, SymbolicValue v2) {
+    return v1.compareTo(v2) > 0 ? new EqualRelation(v2, v1) : new EqualRelation(v1, v2);
+  }
+
+  public static NotEqualRelation notEqual(SymbolicValue v1, SymbolicValue v2) {
+    return v1.compareTo(v2) > 0 ? new NotEqualRelation(v2, v1) : new NotEqualRelation(v1, v2);
+  }
+
+  @Nullable
+  public static Boolean checkValidity(SymbolicValueRelation relation, Collection<SymbolicValueRelation> knownRelations) {
+    Boolean result = checkValidityNoInverse(relation, knownRelations);
+    if (result != null) {
+      return result;
+    }
+    result = checkValidityNoInverse(relation.inverse(), knownRelations);
+    if (Boolean.TRUE.equals(result)) {
+      return Boolean.FALSE;
+    }
+    return null;
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Null is returned but a "Boolean" is expected. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2447)